command_test.go

```diff
@@ -347,3 +347,53 @@ func TestSetHelpCommand(t *testing.T) {
 		t.Errorf("Expected to contain %q message, but got %q", correctMessage, output.String())
 	}
 }
+
+func TestRequiredFlags(t *testing.T) {
+	c := &Command{Use: "c", Run: func(*Command, []string) {}}
+	output := new(bytes.Buffer)
+	c.SetOutput(output)
+	c.Flags().String("foo1", "", "required foo1")
+	c.MarkFlagRequired("foo1")
+	c.Flags().String("foo2", "", "required foo2")
+	c.MarkFlagRequired("foo2")
+	c.Flags().String("bar", "", "optional bar")
+
+	expectedFmt := "Required flags %s have/has not been set"
+	expected := fmt.Sprintf(expectedFmt, fmt.Sprintf("%q, %q", "foo1", "foo2"))
```

Comment:
I think, it would be simpler to write `expected := fmt.Sprintf(Required flags %q, %q have/has not been set, "foo1", "foo2")`